example-lua/src/main/scala/Test.scala

```diff
@@ -0,0 +1,75 @@
+import scalanative.unsafe.*
+import scalanative.unsigned.*
+
+import lua.types.*
+import lua.functions.*
+
+@main def hello =
+  Zone { implicit z =>
+    val stats = alloc[CStruct2[Long, Long]](1)
+
+    val luaAllocator = lua_Alloc.apply {
+      CFuncPtr4.fromScalaFunction { (ud, ptr, osize, nsize) =>
+        val state = ud.asInstanceOf[Ptr[CStruct2[Long, Long]]]
+
+        if nsize == 0.toULong then
+          (!state)._2 = (!state)._2 + osize.toLong
+          scalanative.libc.stdlib.free(ptr)
+          null
+        else
+          (!state)._1 = (!state)._1 + nsize.toLong
+          scalanative.libc.stdlib.realloc(ptr, nsize)
+      }
+    }
```

Comment:
This is really cool - apparently you have to provide your own allocator for Lua: https://www.lua.org/manual/5.4/manual.html#lua_Alloc

Which means you can can control its allocations and track them.